File: horizontalscrollselectedview1/src/main/java/com/example/horizontalscrollselectedview/MainActivity.java
package com.example.horizontalscrollselectedview;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {
    private View leftImageView;
    private View rightImageView;
    private HorizontalSelectedView hsMain;
    private List<String> list = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initView();
        initdates();
    }

    private void initdates() {
        for (int i = 0; i < 6; i++) {
            list.add(i + "0000");
        }
        hsMain.dates(list);
        hsMain.OnSelectList(new HorizontalSelectedView.SelectList() {
            @Override
            public void select(String text) {
                Toast.makeText(MainActivity.this,text,Toast.LENGTH_SHORT).show();
            }
        });
    }

    private void initView() {
        hsMain = (HorizontalSelectedView) findViewById(R.id.hd_main);
        leftImageView = findViewById(R.id.iv_left);
        rightImageView = findViewById(R.id.iv_right);
    }
}

File: kotlinhorizontal/src/main/java/com/example/kotlinhorizontal/MainActivity.kt
package com.example.kotlinhorizontal

import android.support.v7.app.AppCompatActivity
import android.os.Bundle

class MainActivity : AppCompatActivity() {
    private var horizontal: HorizontalScrollView? = null
    private var list = ArrayList<String>()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        horizontal = findViewById(R.id.horizontal_scroll) as HorizontalScrollView

        for (i in 1..10) {
            list!!.add(i.toString() + "000")
        }
        horizontal!!.dates(list!!)
    }
}
